lib/std/fs.zig

```diff
@@ -934,14 +937,16 @@ pub const Dir = struct {
             try os.openatZ(self.fd, sub_path, os_flags, 0);
         errdefer os.close(fd);
 
-        if (!has_flock_open_flags and flags.lock != .None) {
-            // TODO: integrate async I/O
-            const lock_nonblocking = if (flags.lock_nonblocking) os.LOCK_NB else @as(i32, 0);
-            try os.flock(fd, switch (flags.lock) {
-                .None => unreachable,
-                .Shared => os.LOCK_SH | lock_nonblocking,
-                .Exclusive => os.LOCK_EX | lock_nonblocking,
-            });
+        if (builtin.target.os.tag != .wasi) {
+            if (!has_flock_open_flags and flags.lock != .None) {
```

Comment:
```suggestion
        if (!has_flock_open_flags and flags.lock != .None and builtin.target.os.tag != .wasi) {
```

I think we can put all checks in one `if`.

Reply:
yeah this is a little bit trick.. we need to remove out this if block since since the expression is not compiltime-known, and WASI doesn't have `os.flock` that's why I put the OS check prior to the if block. Added comment about this: https://github.com/ziglang/zig/pull/9227/commits/68617c9fb05ad7e92315edb0dec7c5fad56b6265

Reply:
Ah gotcha, makes sense!